src/library_browser.js

```diff
@@ -768,53 +768,46 @@ var LibraryBrowser = {
 
   // TODO: currently not callable from a pthread, but immediately calls onerror() if not on main thread.
   emscripten_async_load_script__deps: ['$UTF8ToString'],
-  emscripten_async_load_script: (url, onload, onerror) => {
-    url = UTF8ToString(url);
-    onload = {{{ makeDynCall('v', 'onload') }}};
-    onerror = {{{ makeDynCall('v', 'onerror') }}};
-
-#if PTHREADS
-    if (ENVIRONMENT_IS_PTHREAD) {
-      err(`emscripten_async_load_script("${url}") failed, emscripten_async_load_script is currently not available in pthreads!`);
-      return onerror ? onerror() : undefined;
-    }
-#endif
-#if ASSERTIONS
-    assert(runDependencies === 0, 'async_load_script must be run when no other dependencies are active');
-#endif
-    {{{ runtimeKeepalivePush() }}}
-
-    var loadDone = () => {
-      {{{ runtimeKeepalivePop() }}}
-      if (onload) {
-        if (runDependencies > 0) {
-          dependenciesFulfilled = onload;
-        } else {
-          onload();
+  emscripten_async_load_script: function(urlPtr, onloadPtr, onerrorPtr) {
+    return new Promise((resolve, reject) => {
+      var url = UTF8ToString(urlPtr);
+
+      if (ENVIRONMENT_IS_PTHREAD) {
+        console.error(`emscripten_async_load_script("${url}") failed, currently not available in pthreads`);
+        if (onerrorPtr) {
+          var onerror = Module['getProcAddress'](onerrorPtr, 'v');
+          onerror();
         }
+        reject(new Error('Not available in pthreads'));
+        return;
       }
-    }
-
-    var loadError = () => {
-      {{{ runtimeKeepalivePop() }}}
-      onerror?.();
-    };
 
-#if ENVIRONMENT_MAY_BE_NODE && DYNAMIC_EXECUTION
-    if (ENVIRONMENT_IS_NODE) {
-      readAsync(url, (data) => {
-        eval(data);
-        loadDone();
-      }, loadError, false);
-      return;
-    }
-#endif
+    
+      var script = document.createElement('script');
+      script.onload = () => {
+        console.log(`Script ${url} loaded successfully.`);
+        if (onloadPtr) {
+          var onload = Module['getProcAddress'](onloadPtr, 'v');
```

Comment:
What is `getProcAddress` here?

Reply:
The getProcAddress function is expected to obtain the address of a web assembly function and return a callable JavaScript version of it.

Reply:
I see.  In that case I would suggest using the `makeDynCall` macro call: `{{{ makeDynCal(...) }}};`

Reply:
Added the macro calls now